src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -349,6 +361,49 @@ impl ConnectionOptionExtracted {
                 },
                 public_keys: None,
             }),
+            CreateConnectionType::MySql => {
+                scx.require_feature_flag(&crate::session::vars::ENABLE_MYSQL_SOURCE)?;
+
+                let cert = self.ssl_certificate;
+                let key = self.ssl_key.map(|secret| secret.into());
+                let tls_identity = match (cert, key) {
+                    (None, None) => None,
+                    (Some(cert), Some(key)) => Some(TlsIdentity { cert, key }),
+                    _ => sql_bail!(
+                        "invalid CONNECTION: both SSL KEY and SSL CERTIFICATE are required"
+                    ),
+                };
+                // Accepts the same SSL Mode values as the MySQL Client
+                // https://dev.mysql.com/doc/refman/8.0/en/connection-options.html#option_general_ssl-mode
+                let tls_mode = match self.ssl_mode.as_ref().map(|m| m.as_str()) {
+                    None | Some("disabled") => MySqlSslMode::Disabled,
+                    // "preferred" intentionally omitted because it has dubious security
+                    // properties.
+                    Some("required") => MySqlSslMode::Required,
+                    Some("verify_ca") | Some("verify-ca") => MySqlSslMode::VerifyCa,
+                    Some("verify_identity") | Some("verify-identity") => {
```

Comment:
I like that you're accepting kebab and snake case here!

Reply:
stolen from the postgres connection implementation!